File: uva_496.py
while True:
    try:
        setA=input().split()
        setB=input().split()
    except:
        break
    countA, countB=0, 0
    for itemA in setA:
        for itemB in setB:
            if itemA==itemB:
                countA+=1
                break
    for itemB in setB:
        for itemA in setA:
            if itemB==itemA:
                countB+=1
                break
    if countA==len(setA) and len(setA)<len(setB):
        print('A is a proper subset of B')
    elif countB==len(setB) and len(setB)<len(setA):
        print('B is a proper subset of A')
    elif not countA and not countB:
        print('A and B are disjoint')
    elif countA==len(setA) and countB==len(setB) and len(setA)==len(setB):
        print('A equals B')
    else:
        print('I\'m confused!')

File: uva_496.cpp
# include <string>
# include <sstream>
# include <iostream>
using namespace std;
int main(void){
    string lineA, lineB;
    while(getline(cin, lineA)){
        getline(cin, lineB);
        int setA[100], setB[100];
        int countA=0, totalA=-1, countB=0, totalB=-1;
        stringstream ss;
        ss<<lineA;
        while(ss>>setA[++totalA]);
        ss.str("");
        ss.clear();
        ss<<lineB;
        while(ss>>setB[++totalB]);
        for (int i=0; i<totalA; ++i)
            for (int j=0; j<totalB; ++j)
                if (setA[i]==setB[j]){
                    ++countA;
                    break;
                }
        for (int i=0; i<totalB; ++i)
            for (int j=0; j<totalA; ++j)
                if (setB[i]==setA[j]){
                    ++countB;
                    break;
                }
        if (countA==totalA && totalA<totalB)
            cout << "A is a proper subset of B\n";
        else if (countB==totalB && totalB<totalA)
            cout << "B is a proper subset of A\n";
        else if (!countA && !countB)
            cout << "A and B are disjoint\n";
        else if (countA==totalA && countB==totalB && totalA==totalB)
            cout << "A equals B\n";
        else
            cout << "I'm confused!\n";
    }
    return 0;
}